### PR TITLE
Update Go to 1.14 and example app to Flink 1.10

### PR DESCRIPTION
**Pull request by yuchaoran2011**

### CODE REVIEW

**Comment by yuchaoran2011** [comment]
`make lint` failed with a timeout error.
```
ERRO Timeout exceeded: try increase it by passing --timeout option 
```
Maybe a transient thing in Travis. I have the same version of `golangci-lint` and lint worked on my machine. If an admin can restart the build and try again, it should work.

**Comment by mwylde** [comment]
I tried restarting it and still got a timeout. Can you try increasing the timeout in the travis config in this branch?

**Comment by yuchaoran2011** [comment]
@mwylde I just increased the timeout and it worked :)

**Comment by glaksh100** [comment]
@yuchaoran2011 Thanks a lot for contributing this change! I wanted to give you an update on the status of merging this PR — we're trying to come up with a standardized way to test Go version upgrades (hence the delay in merging this). I'll update this PR with the plan once it's ready.

**Comment by yuchaoran2011** [comment]
@glaksh100 Thanks. Sounds good!

**Comment by yuchaoran2011** [comment]
Hi @glaksh100, any updates on Go upgrade? We would really like to see the base image upgraded to Go 1.14 because earlier versions pose some security vulnerabilities for our use case. We could fork the project and then build using our own Dockerfile. But we want to avoid doing that to due to the burden of maintaining our own fork (e.g. sync with upstream etc).

File: integ/utils/utils.go [hunk 1]
```diff
@@ -92,7 +92,8 @@ func (f *TestUtil) Cleanup() {
 			log.Errorf("Failed to fetch flink apps during cleanup: %v", err)
 		} else {
 			// make sure none of the apps have left-over finalizers
-			for _, app := range flinkApps.Items {
+			for i := range flinkApps.Items {
+				app := flinkApps.Items[i]
```
Comment (yuchaoran2011) [inline integ/utils/utils.go:96]: Without this change, the output of `make lint` will say:
```
integ/utils/utils.go:98:34: G601: Implicit memory aliasing in for loop. (gosec)
					_, _ = f.FlinkApps().Update(&app)
					                            ^
```

File: integ/checkpoint_failure_test.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (s *IntegSuite) TestJobWithTaskFailures(c *C) {
 func (s *IntegSuite) TestCheckpointTimeout(c *C) {
 	failingJobTest(s, c, "checkpointtimeout", func() {
 		// cause checkpoints to take 120 seconds
-		err := ioutil.WriteFile(s.Util.CheckpointDir+"/checkpoint_delay", []byte("120000"), 0644)
+		err := ioutil.WriteFile(s.Util.CheckpointDir+"/checkpoint_delay", []byte("120000"), 0600)
```
Comment (yuchaoran2011) [inline integ/checkpoint_failure_test.go:81]: Without this change, `make lint` will fail with:
```
integ/checkpoint_failure_test.go:81:10: G306: Expect WriteFile permissions to be 0600 or less (gosec)
		err := ioutil.WriteFile(s.Util.CheckpointDir+"/checkpoint_delay", []byte("120000"), 0644)
		       ^
```

**Pull request closed without merging**